### ISSUE TITLE
Exploiters being able to mess up powerups

### ISSUE THREAD

**Issue by Proxofer** [ipodtouch0218/NSMB-MarioVsLuigi#101]
as i mentioned on my other comment, i also noticed that the exploiter that cloned their character multiple times also made himself invincible and was able to randomize the powerups in a loop.

**Comment by Mardz38** [comment]
Don't play in public games then.

**Comment by Proxofer** [comment]
sure thing sherlock, that would make the game a lot better!

**Comment by Mardz38** [comment]
Here's a better idea smartass, don't play the game if you can't deal with hackers. They're in pretty much every online game.

**Comment by abso1utezer0** [comment]
> Here's a better idea smartass, don't play the game if you can't deal with hackers. They're in pretty much every online game.

This is a horrible mindset, it's up to the developers to combat hacking, and the main developer has put some measures in place. @Proxofer, please join the [Discord server](https://discord.com/invite/dgKVaUKpj5) and report any hackers in the respective channel. They will be permanently banned from playing the game if all goes well.

Though, I think much more should be done to combat this issue. Some sort of checking should be implemented, because it really should not be possible for someone to move past the max speed limit or spawn hundreds of items without the other clients catching on and kicking/banning the user from the lobby. A host should also be able to kick/ban users mid-game through the leaderboard UI. I hope some better anti-cheating measures are developed soon, and I am sorry for the hostile reply you got originally, Proxofer.

**Comment by Mardz38** [comment]
The funny thing is that those messages were made before the channel even existed.

**Comment by ipodtouch0218** [comment]
While this was a thing, it only happens now when the player is the room host. The problem is, there's nothing keeping the cheater from stealing room host.

This will be completely patched in 1.8 with the networking changes.

**Comment by PoketrainerMS** [comment]
some info ive gathered about this by watching my friend make a mod for this game
("A" is modder, "M" is a friend who helped test, i silently watched and asked some questions in vc, if you're in the discord server they did this in, you know who im talking about, though their names dont matter so im not bothering typing it all out)

so, A was messing around with coin reward rng, and changed it to be only megamush to test the limits of what he can change about the game, and how it looks to someone without the same build. A loaded up a lobby, and had M join (note: M does not have the custom build). A got a megamush, and on M's screen, A was just normal luigi, but had the hitbox/hurtbox of megamush, and i dont remember which it was, but either A forever had the dust effects of the megamush after or just kept the megamush hitbox (no hurtbox). when M later got a powerup, for A it showed up as him having megamush, even after losing the powerup he thinks he got. the hitbox went back to normal after the powerup was lost, but M could still kill A from a distance (megamush hurtbox).

basically, tl;dr/conclusion:
modders can change what they get and it shows up weird to other players without the same build, and can lead to unfair gameplay (or basically hacking the game for wins if in the wrong hands)

so uhhhh
if you mod the game, keep it in private servers lol (thats what A does)

**Comment by ipodtouch0218** [comment]
@PoketrainerMS Yeah, it's a problem with PUN having each user have authority over their own character. Come v1.8.0.0-beta, we're changing backends to have full host authority.

**Comment by SLG64** [comment]
When I first started, there was a pretty big hacker. I can't remember their username of course, but I can understand the irritation from those kinds of players. I do remember a Player3#### hacking in all the lobbies I was in and trying to talk to only me, which felt weird of course. They would take the host over and kick everyone but me, and that's the unsettling part. So here is my questions; How are hackers able to do this and how do we stop them? Are we going top have to constantly report them? Will there be any protection against this in the future?
